From a74af39989a4b9c4cd57658e62feb225580843a1 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Tue, 4 Oct 2016 14:57:44 -0700
Subject: [PATCH] Skip fingerprint check when AB updates are enabled.

There's currently no good way to read the ro.build.fingerprint off the inactive
system partition from within the updater script. This is only used for
incremental OTA updates.

It's /okay/ to skip this sanity check because:
1) The updater will have already verified the version before downloading
2) If the fingerprint mismatches, the incremental OTA will simply fail and the
   system will recover by synchronizing the inactive partitions with the active
   partitions

---
 tools/releasetools/edify_generator.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/releasetools/edify_generator.py b/tools/releasetools/edify_generator.py
index 89840c6..354ad26 100644
--- a/tools/releasetools/edify_generator.py
+++ b/tools/releasetools/edify_generator.py
@@ -86,6 +86,7 @@ class EdifyGenerator(object):
     if not fp:
       raise ValueError("must specify some fingerprints")
     cmd = (
+           ' ||\n    getprop("ro.silk.update.ab") == "true" ||\n' +
            ' ||\n    '.join([('getprop("ro.build.fingerprint") == "%s"')
                         % i for i in fp]) +
            ' ||\n    abort("Package expects build fingerprint of %s; this '
-- 
2.9.2

